ingeniamotion/configuration.py

```diff
@@ -712,3 +736,145 @@ def restore_tcp_ip_parameters(self, servo: str = DEFAULT_SERVO) -> None:
         """
         drive = self.mc._get_drive(servo)
         drive.restore_tcp_ip_parameters()
+
+    def get_drive_info_coco_moco(
+        self, alias: str
+    ) -> tuple[list[Optional[int]], list[Optional[int]], list[Optional[str]], list[Optional[int]]]:
+        """Get info from COCO and MOCO registers.
+
+        Args:
+            alias: Servo alias.
+
+        Returns:
+            Product codes (COCO, MOCO).
+            Revision numbers (COCO, MOCO).
+            FW versions (COCO, MOCO).
+            Serial numbers (COCO, MOCO).
+
+        """
+        prod_codes: list[Optional[int]] = [None, None]
+        rev_numbers: list[Optional[int]] = [None, None]
+        fw_versions: list[Optional[str]] = [None, None]
+        serial_number: list[Optional[int]] = [None, None]
+
+        for subnode in [0, 1]:
+            # Product codes
+            try:
+                prod_codes[subnode] = self.get_product_code(alias, subnode)
+            except (
+                ILError,
+                IMException,
+            ) as e:
+                self.logger.error(e)
+            # Revision numbers
+            try:
+                rev_numbers[subnode] = self.get_revision_number(alias, subnode)
+            except (ILError, IMException) as e:
+                self.logger.error(e)
+            # FW versions
+            try:
+                fw_versions[subnode] = self.get_fw_version(alias, subnode)
+            except (ILError, IMException) as e:
+                self.logger.error(e)
+            # Serial numbers
+            try:
+                serial_number[subnode] = self.get_serial_number(alias, subnode)
+            except (ILError, IMException) as e:
+                self.logger.error(e)
+
+        return prod_codes, rev_numbers, fw_versions, serial_number
+
+    def get_drive_info(self, alias: str) -> tuple[int, int, str, int]:
```

Comment:
I'm not sure that this function has sense in ingeniamotion, this will be only used in ML3, in my opinion should be removed